edurange_refactored/user/views.py

```diff
@@ -261,23 +263,39 @@ def instructor():
 
     return redirect(url_for("dashboard.instructor"))
 
-
 @blueprint.route("/notification", methods=["GET", "POST"])
 @login_required
 def notification():
     if request.method == "POST":
         process_request(request.form)
 
-    notificationList = Notification.query.all()
+    notifications_raw = Notification.query.all()
+    notifications_serialized = []
+
+    for notif in notifications_raw:
+        notif_serialized = {}
+        for key, value in notif.__dict__.items():
+            if key != '_sa_instance_state':
+                if isinstance(value, datetime):
+                    value = value.isoformat()
+                notif_serialized[key] = value
+        notifications_serialized.append(notif_serialized)
+    
+    for index, notif in enumerate(notifications_serialized):
```

Comment:
What is this loop used for? I'm not super familiar with how locals() works, but it doesn't seem like those new variables get used.